instrumentation/servlet/servlet-common/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/servlet/InjectionState.java

```diff
@@ -8,9 +8,11 @@
 // this is shared by both ServletOutputStream and PrintWriter injection
 public class InjectionState {
   private static final int HEAD_TAG_WRITTEN_FAKE_VALUE = -1;
-  private static final int HEAD_TAG_LENGTH = "<head>".length();
+  private static final String HEAD_PREFIX = "<head";
+  private static final StringBuilder HEAD_PREFIX_BUILDER = new StringBuilder("");
```

Comment:
Having a mutable global variable is bound to fail in a multi-threaded scenario. If the instrumented app handles concurrent requests you will run into races and concurrent modifications.

Reply:
Change another way to do that